components/VM_Arm/src/main.c

```diff
@@ -802,7 +804,17 @@ static int generate_fdt(vm_t *vm, void *fdt_ori, void *gen_fdt, int buf_size, si
     }
 
     if (config_set(CONFIG_VM_PCI_SUPPORT)) {
-        err = fdt_generate_vpci_node(vm, pci, gen_fdt, GIC_IRQ_PHANDLE);
+        int gic_offset = fdt_path_offset(fdt_ori, GIC_NODE_PATH);
```

Comment:
Should we use `vm.gic_path` here? The file gets initialized in the other code change you did, so this init might have to happen a bit earlier then.

Reply:
Order was okay. vm.gic_path is set before this function is called